pkg/controller/clusterstorage/clusterstorage_controller.go

```diff
@@ -183,7 +184,36 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
-	// StorageClass already exists - don't requeue
+	// Check to see if modifications have been made to the StorageClass attributes
+	needUpdate := false
+	comparisonSC := sc.DeepCopy()
+	comparisonSC.ObjectMeta = found.ObjectMeta
+
+	// Define a default ReclaimPolicy for comparison
+	if comparisonSC.ReclaimPolicy == nil {
+		deletePolicy := corev1.PersistentVolumeReclaimDelete
+		comparisonSC.ReclaimPolicy = &deletePolicy
+	}
+
+	if !reflect.DeepEqual(comparisonSC, found) {
+		needUpdate = true
+	}
+
+	// If a change has been detected, update the StorageClass.
+	if needUpdate {
+		reqLogger.Info("StorageClass already exists and needs to be updated", "StorageClass.Name", found.Name)
+		err = r.client.Update(context.TODO(), sc)
```

Comment:
I think you want to use `comparisonSC` rather than `sc`, right?

Reply:
Previously, when we were using `Create`, this failed because we had previously copied over the `ObjectMeta`. Updating this line to `comparisonSC` updates the full ObjectMeta without issue. I went ahead and made this change.